### PR TITLE
Add checks for assign, unassign and reassign commands

### LINKED ISSUE #55: Add checks for assign command
A worker should only be allowed to take on a role in a shift if:

- [x] That role is present in the worker's role set
- [x] The worker is available and not on leave for that shift
- [x] That role needs to be filled in the shift (ie. that role is present in the shift's role requirement set and the quantity of workers needed for that role has not been fulfilled)

### PR DESCRIPTION
**Pull request by wakululuu**
Resolves #55

### CODE REVIEW

File: src/main/java/seedu/address/model/shift/RoleRequirement.java [hunk 1]
```diff
@@ -67,24 +105,26 @@ public boolean isSameRoleRequirement(RoleRequirement otherRoleRequirement) {
...
      * This defines a stronger notion of equality between two role requirements.
      */
     @Override
     public boolean equals(Object other) {
         return other == this
                 || (other instanceof RoleRequirement
                 && role.equals(((RoleRequirement) other).role)
-                && quantity == ((RoleRequirement) other).quantity);
+                && quantityRequired == ((RoleRequirement) other).quantityRequired
+                && quantityRequired == ((RoleRequirement) other).quantityRequired);
```
Comment (tnsyn) [inline src/main/java/seedu/address/model/shift/RoleRequirement.java:121]: Possible duplication?

File: src/main/java/seedu/address/model/shift/RoleRequirement.java [hunk 2]
```diff
@@ -49,16 +75,28 @@ public static boolean isValidRoleRequirement(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if the quantity filled has not reached the quantity required.
+     */
+    public boolean isFilled() {
+        assert quantityFilled <= quantityRequired;
```
Comment (tnsyn) [inline src/main/java/seedu/address/model/shift/RoleRequirement.java]: Would it be more meaningful to put this assertion in the constructor instead? Just a suggestion, not sure if it's good practice

**Review by tnsyn — COMMENTED** [review]
LGTM, just a couple of very small comments only!

File: src/main/java/seedu/address/model/shift/RoleRequirement.java (hunk 1 above)
Comment (wakululuu) [reply src/main/java/seedu/address/model/shift/RoleRequirement.java:121]: Good catch, thank you!

File: src/main/java/seedu/address/model/shift/RoleRequirement.java (hunk 2 above)
Comment (wakululuu) [reply src/main/java/seedu/address/model/shift/RoleRequirement.java]: Makes sense, edited

**Review by tnsyn — APPROVED** [review]

**Pull request merged**